src/node/eviction.cpp

```diff
@@ -204,7 +213,8 @@ void ProtectEvictionCandidatesByRatio(std::vector<NodeEvictionCandidate>& evicti
     // or disadvantaged characteristics.
     ProtectEvictionCandidatesByRatio(vEvictionCandidates);
 
-    if (vEvictionCandidates.empty()) return std::nullopt;
+    if (vEvictionCandidates.empty())
+        return force ? force_evict : std::nullopt;
```

Comment:
Since `force_evict` is already an `optional` defaulting to `nullopt`, this can be simplified in conjunction with my other suggestion.
```suggestion
    if (vEvictionCandidates.empty()) return force_evict;
```